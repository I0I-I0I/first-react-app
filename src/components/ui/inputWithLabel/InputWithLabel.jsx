import { useEffect, useRef } from "react";
import styles from "./InputWithLabel.module.css";

const InputWithLabel = ({
	id,
	value,
	type = "text",
	onInputChange,
	isFocused,
	children,
}) => {
	const inputRef = useRef();

	useEffect(() => {
		if (isFocused && inputRef.current) {
			inputRef.current.focus();
		}
	}, [isFocused]);

	return (
		<>
			<label className={styles.label} htmlFor={id}>
				{children}
			</label>
			&nbsp;
			<input
				placeholder="Enter something..."
				className={styles.input}
				ref={inputRef}
				id={id}
				type={type}
				value={value}
				onChange={onInputChange}
			/>
		</>
	);
};

export default InputWithLabel;
