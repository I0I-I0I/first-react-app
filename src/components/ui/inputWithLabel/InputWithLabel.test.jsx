import { fireEvent, render, screen } from "@testing-library/react";
import { describe, expect, it, vi } from "vitest";
import InputWithLabel from "./InputWithLabel.jsx";

const props = {
	id: "input1",
	value: "React",
	type: "text",
	onInputChange: vi.fn(),
	isFocused: false,
	children: "Search",
};

describe("InputWithLabel", () => {
	it("correct input value", () => {
		render(<InputWithLabel {...props} />);
		const input = screen.getByRole("textbox");
		expect(input.value).toEqual("React");
	});

	it("text in label", () => {
		render(<InputWithLabel {...props} />);
		const label = screen.getByLabelText("Search");
		expect(label).toBeDefined();
	});

	it("change function is called once", () => {
		const fn = vi.fn();
		render(<InputWithLabel {...props} onInputChange={fn} />);
		const input = screen.getByRole("textbox");
		fireEvent.change(input, {
			target: { value: "ForTest" },
		});
		expect(fn).toHaveBeenCalledOnce();
	});

	it("renders snapshot", () => {
		const { container } = render(<InputWithLabel {...props} />);
		expect(container.firstChild).toMatchSnapshot();
	});
});
