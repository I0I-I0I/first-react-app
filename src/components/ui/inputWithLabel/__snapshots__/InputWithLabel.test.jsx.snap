// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`InputWithLabel > renders snapshot 1`] = `
<label
  class="_label_1ba982"
  for="input1"
>
  Search
</label>
`;
